### PR TITLE
DROOLS-3287: [DMN Designer] Enable DMN editor by default in distributions

### PR DESCRIPTION
**Pull request by danielzhe**
This PR requires:

[kiegroup/kie-wb-distributions#848](https://github.com/kiegroup/kie-wb-distributions/pull/848)

### CODE REVIEW

**Comment by danielzhe** [comment]
Jenkins execute full downstream build

File: kie-wb-common-dmn/kie-wb-common-dmn-project-client/src/main/java/org/kie/workbench/common/dmn/project/client/handlers/DMNDiagramNewResourceHandler.java [hunk 1]
```diff
@@ -58,16 +51,6 @@ public DMNDiagramNewResourceHandler(final DefinitionManager definitionManager,
               projectDiagramServices,
               busyIndicatorView,
               projectDiagramResourceType);
-        this.authorizationManager = authorizationManager;
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-project-client/src/main/java/org/kie/workbench/common/dmn/project/client/handlers/DMNDiagramNewResourceHandler.java]: Can you also remove the unused `authorizationManager` and `sessionInfo` parameters from the constructor.

**Comment by danielzhe** [comment]
@manstis I was not sure about what revert, since the PR says that "integrates editor into workbench". So I thought that if I revert more I will break the integration and not just enabling the DMN as default.

**Comment by danielzhe** [comment]
@manstis I did the required changes.

File: kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/authz/PermissionTreeSetup.java [hunk 2]
```diff
@@ -131,8 +131,6 @@ public void configureTree() {
                                           i18n.XLSScoreCard());
         editorTreeProvider.registerEditor(STUNNER_DESIGNER,
                                           i18n.StunnerDesigner());
-        editorTreeProvider.registerEditor(DMN_DESIGNER,
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/authz/PermissionTreeSetup.java]: I think there's a [test](https://github.com/kiegroup/kie-wb-common/blob/master/kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/test/java/org/kie/workbench/common/workbench/client/authz/PermissionTreeSetupTest.java#L172) that is going to fail.... but let's wait for Jenkins.

**Comment by danielzhe** [comment]
Jenkins execute full downstream build

File: kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/authz/PermissionTreeSetup.java (hunk 2 above)
Comment (jomarko) [reply kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/authz/PermissionTreeSetup.java]: +1

**Comment by danielzhe** [comment]
Jenkins execute full downstream build

**Comment by danielzhe** [comment]
@jomarko Those files are updated in [this PR](https://github.com/kiegroup/kie-wb-distributions/pull/848), since is another repository.

**Comment by jomarko** [comment]
@danielzhe ok, sorry, I missed it. May I ask you to mention if jira is solved by an ensemble? It is used often in kiegroup, one example:
https://github.com/kiegroup/appformer/pull/493#issue-222240553

File: kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/authz/PermissionTreeSetup.java (hunk 2 above)
Comment (danielzhe) [reply kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/authz/PermissionTreeSetup.java]: Test updated.

File: kie-wb-common-dmn/kie-wb-common-dmn-project-client/src/main/java/org/kie/workbench/common/dmn/project/client/handlers/DMNDiagramNewResourceHandler.java [hunk 3]
```diff
@@ -20,54 +20,32 @@
...
-
     protected DMNDiagramNewResourceHandler() {
         this(null,
              null,
              null,
-             null,
-             null,
-             null);
+             null
+        );
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-project-client/src/main/java/org/kie/workbench/common/dmn/project/client/handlers/DMNDiagramNewResourceHandler.java]: Sorry, minor niggle, could you please correct this code formatting? (closing brace on the same line as the last parameter).

Comment (danielzhe) [reply kie-wb-common-dmn/kie-wb-common-dmn-project-client/src/main/java/org/kie/workbench/common/dmn/project/client/handlers/DMNDiagramNewResourceHandler.java]: Done.

**Pull request merged**